models/base.py

```diff
@@ -430,7 +443,12 @@ def get_name_in_current_ctx(self) -> str:
     cur_scope_abs = cur_scope.get_abs_name_ctx_list()
     self_name_abs = self.get_abs_name_ctx_list()
     assert cur_scope_abs[0] is self_name_abs[0]  # same root
-    assert len(cur_scope_abs) >= len(self_name_abs)  # not implemented otherwise
+    if len(cur_scope_abs) < len(self_name_abs): # grabbing a sublayer
```

Comment:
Better do it the way as I suggested.
I.e. first:
`max_common_len = max(len(cur_scope_abs), len(cur_scope_abs))`
Then for the `while` loop below, add also `common_len < max_common_len and ` in the beginning.
And then add your code below for the case `common_len < len(self_name_abs)`.

Reply:
Ahh just commited another change before I saw your comment :D, but sure we can do it like this, let me restructe the code.

Reply:
This crashes with index out of range though. Should it be `common_len+1 < max_common_len` ? Or we need to change the second part of the while statement.

Reply:
It's very possible that I have made some +- 1 or 2 error here. :) It's a bit late and I'm going to sleep now.

Reply:
Same :D, lets talk about this tomorrow/Monday